pyproject.toml

```
@@ -854,7 +853,12 @@ ignore = [
     "COM819",
 
     # Disabled because ruff does not understand type of __all__ generated by a function
-    "PLE0605"
+    "PLE0605",
+
+    "TC006", # disabled temporarily on ruff 0.10.0 update
```

I think `TC006` should be disabled. It only really makes sense for highly performance critical code but those cases are often better served by just moving the `cast` call into a `if TYPE_CHECKING` block outright. Something we already do in the relevant cases

https://github.com/home-assistant/core/blob/cdead8661d7c0a8fbdfd51f3b5039af5b1d30a21/homeassistant/core.py#L751-L765

A suggestion:

```suggestion
    # Quotes for typing.cast generally not necessary, only for performance critical paths
    "TC006", # Add quotes to type expression in typing.cast()
```